pkg/drivers/vfkit/vfkit.go

```diff
@@ -286,12 +290,23 @@ func (d *Driver) setupIP(mac string) error {
 		time.Sleep(2 * time.Second)
 	}
 
-	if err != nil {
-		return fmt.Errorf("IP address never found in dhcp leases file %v", err)
+	if err == nil {
+		log.Debugf("IP: %s", d.IPAddress)
+		return nil
 	}
-	log.Debugf("IP: %s", d.IPAddress)
+	if !isBootpdError(err) {
+		return errors.Wrap(err, "IP address never found in dhcp leases file")
+	}
+	if unblockErr := firewall.UnblockBootpd(); unblockErr != nil {
+		klog.Errorf("failed unblocking bootpd from firewall: %v", unblockErr)
+		exit.Error(reason.IfBootpdFirewall, "ip not found", err)
```

Comment:
This uses sudo with a password, blocking minikube for ever when running it by another program.

I don't think we should use such helpers. If minikube needs some setup as root, it should be done outside of minikube. If we want to make this easy, we cna provide a minikube setup command that will use sudo.

Another minor issue, this shows a bogus message about socket_vmnet which is not used by vfkit.

When this does not work, (e.g. I tried to change the virtio-net device to use gvproxy), this ends in very bad user experience - minikube blocks for long time on sudo, then try to restart the vm, finaly it fail after many minutes. It is better to do less and fail fast instead of trying to fix something we are not sure about.

I guess the same issue exist with qemu driver using socket_vmnet, so this is was not introduced by this driver.

Reply:
It was added (from qemu) after feedback about not working (dhcp not found):
https://github.com/kubernetes/minikube/pull/19423#issuecomment-2292431081

Reply:
I'm not a big fan of the "retry" myself, I would prefer it to just fail